_content/network/account/implement.md

```diff
@@ -0,0 +1,117 @@
+# Implementing you own Account Server for application management
+
+To implement your own account server, you need to implement a couple of things:
+
+1. An access key validation endpoint
+2. Endpoints to generate and validate application access tokens
+
+# 1. Access Key validation
+
+To validate access keys, MQTT needs to know what rights the access key provides.
+Since the access keys are opaque, it consults an endpoint to fetch those rights:
+
+```plaintext
+GET /api/v2/applications/{app_id}/rights
+Authorization: Key {app_access_key}
+```
+
+If the app id or access key is invalid, it should return `401`.
+
+When successful, the enpoint should return:
+
+```plaintext
+200 OK
+Content-Type: application/json
+
+[
+  "messages:up:r",
+  "messages:down:w"
+]
+```
+
+The meaning of rights returned here are:
+
+- `messages:up:r`:  The right to read the uplink messages from devices an application
+- `messages:down:w`: The right to send the downlink messages to devices of an application
+
+The presence of these rights tells the MQTT plugin what type of messages allow
+the client to see or send.
+
+> Important Note: due to the limited right management of most MQTT implementations, make
```

Comment:
I guess most MQTT brokers support read and write rights. 

For example, for Mosquitto:

```c
/*
 * Function: mosquitto_auth_acl_check
 *
 * Called by the broker when topic access must be checked. access will be one
 * of MOSQ_ACL_READ (for subscriptions) or MOSQ_ACL_WRITE (for publish). Return
 * MOSQ_ERR_SUCCESS if access was granted, MOSQ_ERR_ACL_DENIED if access was
 * not granted, or MOSQ_ERR_UNKNOWN for an application specific error.
 */
int mosquitto_auth_acl_check(void *user_data, const char *clientid, const char *username, const char *topic, int access);
```

Similarly, for RabbitMQ:

> RabbitMQ distinguishes between configure, write and read operations on a resource. The configure operations create or destroy resources, or alter their behaviour. The write operations inject messages into a resource. And the read operations retrieve messages from a resource.